src/controlplane/client/connector.go

```diff
@@ -0,0 +1,294 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+)
+
+const (
+	DefaultTimout      = 5000 * time.Millisecond
+	defaultMetricsPath = "/metrics"
+	mTLSAuth           = "mTLS"
+	bearerAuth         = "bearer"
+)
+
+// Connector provides an interface to retrieve []connParams to connect to a Control Plane instance.
+type Connector interface {
+	Connect() (*connParams, error)
+}
+
+type defaultConnector struct {
+	// TODO: Use a non-sdk logger
+	logger          log.Logger
+	kc              kubernetes.Interface
+	inClusterConfig *rest.Config
```

Comment:
By seeing this I get the feel that the connector is doing too many things: Checking connectivity and handling authentication. Perhaps we could build a separate object, some sort of "authenticator"? This object could receive these two args, and together with the `auth` section of the config, take care of authenticating requests. WDYT?